lib/n3h-mod-dht-fullsync/dht-fullsync.js

```diff
@@ -335,54 +342,54 @@ class DhtBackendFullsync extends AsyncClass {
       }
     }
 
-    // next peer
-    if (!this._gossipStack.length) {
-      ++this._lastGossipIdx
-
-      if (!this._peerList || this._lastGossipIdx >= this._peerList.length) {
-        // either we have no peers
-        // or we just wrapped our peer list
-        // wait for half a second
-        this._lastGossipIdx = -1
-        this._gossipStack.push({
-          type: 'wait',
-          until: Date.now() + 500
-        })
-        return
-      }
-
-      const peerAddress = this._peerList[this._lastGossipIdx]
-
-      // skip if this peer is us :)
-      if (peerAddress === this._thisPeer.peerAddress) {
-        return setImmediate(() => this._gossip())
-      }
+    // _gossipStack must be empty before gossiping new
+    if (this._gossipStack.length) {
+      return
+    }
 
-      const msgId = this.$createUid()
+    // -- gossip locHashes to next peer -- //
 
-      this._spec.$emitEvent(DhtEvent.gossipTo(
-        [peerAddress], gossip.locHashes(msgId, this._locHashes)))
+    ++this._lastGossipIdx
 
-      // give this peer 3 seconds
-      // if the whole back-n-forth takes longer,
-      // then they are too laggy, move on to the next
+    if (!this._peerList || this._lastGossipIdx >= this._peerList.length) {
+      // either we have no peers or we just wrapped our peer list
+      // wait for half a second
+      this._lastGossipIdx = -1
       this._gossipStack.push({
-        type: 'waitPeer',
-        until: Date.now() + 3000,
-        peerAddress,
-        msgId
+        type: 'wait',
+        until: Date.now() + 500
       })
+      return
+    }
+
+    const peerAddress = this._peerList[this._lastGossipIdx]
+
+    // skip if this peer is us :)
+    if (peerAddress === this._thisPeer.peerAddress) {
+      return setImmediate(() => this._gossip())
```

Comment:
do we want to delete this since we're not using the `_thisPeer` really? Can do it later, doesn't have to be in this PR

Reply:
I'd rather try a getDhtRef() function and init the DHT with a correct _thisPeer.
Also wouldn't we want to have a _dht per dnaAddress ?